internal/dataplane_controller/dataplane_controller.go

```diff
@@ -36,30 +37,41 @@ type DataPlaneReconciler struct {
 	Log    logr.Logger
 	Scheme *runtime.Scheme
 	// reconcile time duration, defaults to 10s
-	ReconcileWait time.Duration
-	Recorder      record.EventRecorder
-	Predicates    predicate.Predicate
-	NgStore       store.Store
-	CustomerName  string
-	EnablePrivate bool
+	ReconcileWait   time.Duration
+	Recorder        record.EventRecorder
+	Predicates      predicate.Predicate
+	NgStore         store.Store
+	CustomerName    string
+	EnablePrivate   bool
+	InClusterClient client.Client
 }
 
 func NewDataplaneReconciler(mgr ctrl.Manager, enablePrivate bool, customerName string) *DataPlaneReconciler {
 	initLogger := ctrl.Log.WithName("controllers").WithName("dataplane")
+	inClusterClient, err := getInClusterClient()
+	if err != nil {
+		fmt.Println("failed to load incluster client, private_mode won't work, reason: ", err.Error())
+	}
```

Comment:
log error and return err

Reply:
this will only run once in the controller startup, hence added panic